frontend/src/app/admin/Organization/TenantDataAccessFormContainer.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import { toast } from "react-toastify";
+import {
+  useAppInsightsContext,
+  useTrackEvent,
+} from "@microsoft/applicationinsights-react-js";
+import { Redirect } from "react-router-dom";
+
+import Alert from "../../commonComponents/Alert";
+import { showNotification } from "../../utils";
+
+import { OrganizationOptions } from "./OrganizationDropDown";
+import TenantDataAccessForm from "./TenantDataAccessForm";
+
+export const GET_ORGANIZATIONS_QUERY = gql`
+  query GetOrganizations {
+    organizations {
+      externalId
+      name
+    }
+  }
+`;
+
+export const SET_TENANT_DATA_ACCESS = gql`
+  mutation SetCurrentUserTenantDataAccessOp(
+    $organizationExternalId: String
+    $justification: String
+  ) {
+    setCurrentUserTenantDataAccess(
+      organizationExternalId: $organizationExternalId
+      justification: $justification
+    ) {
+      id
+      email
+      permissions
+      role
+      organization {
+        name
+        externalId
+      }
+    }
+  }
+`;
+
+const TenantDataAccessFormContainer: any = () => {
+  const [submitted, setSubmitted] = useState(false);
+  const { data, loading, error } = useQuery<OrganizationOptions, {}>(
+    GET_ORGANIZATIONS_QUERY,
+    {
+      fetchPolicy: "no-cache",
+    }
+  );
+  const appInsights = useAppInsightsContext();
+  const [setTenantDataAccess] = useMutation(SET_TENANT_DATA_ACCESS);
+  const trackSaveSettings = useTrackEvent(appInsights, "Save Settings", null);
+
+  if (loading) {
+    return <p> Loading... </p>;
```

Comment:
**suggestion (non-blocking):** Is it worth using `LoadingCard` here instead?

Reply:
That looks nicer : )